### DISCUSSION TITLE
Password Input Field Not Masking Text on iOS

### DISCUSSION THREAD

**Discussion by OiviaDesu** [ppy/osu#25487] [Ideas] [answered]
![IMG_4101](https://github.com/ppy/osu/assets/109123984/67f1c1df-7ebf-4b3e-8b3c-5dbd89483187)

## Description
I've encountered an issue with the `LoginForm` in the `osu.Game.Overlays.Login` namespace where the password input field is not masking the input text when using a third-party keyboard on iOS. This means that the typed password is visible in the suggestion field of the keyboard, which is a security concern.

## Expected Behavior
When typing into the password field, the input text should be masked (e.g., replaced with dots or asterisks) to prevent it from being visible. Additionally, third-party keyboards should be disallowed for password input fields on iOS.

## Actual Behavior
The typed password is visible in the suggestion field of the keyboard when using a third-party keyboard on iOS. The system does not automatically switch to the system keyboard when the password field is focused.

## Steps to Reproduce
1. Open the login form on an iOS device with a third-party keyboard installed and enabled. Im using Microsoft SwiftKey 
2. Tap on the password input field.
3. Type into the password field using the third-party keyboard. For example, Im typing "ThisShouldBeHidden" in the Screenshot
## Environment
- OS: iOS 17
- Version: 2023.1114.1-lazer

## Additional Information
This issue seems to be related to the `TextBox` control used for the password input field. A potential solution might be to implement a custom control that masks the input text and disallows third-party keyboards.

**Comment by peppy** [comment, marked as answer]
https://github.com/ppy/osu-framework/issues/6010
https://github.com/ppy/osu/issues/24041
https://github.com/libsdl-org/SDL/issues/7965
